### ISSUE TITLE
Problem Using `apt` After Installing Patch

### ISSUE THREAD

**Issue by S4JJ4D** [mathworks/build-glibc-bz-19329-patch#8]
Running `Ubuntu 20.04.4 LTS x86_64`.

**Before Installation**:
```
$ dpkg-query --show libc6:amd64

libc6:amd64	2.31-0ubuntu9.7
```
I couldn't find releases for 9.7 version, so I downloaded pre-built packages for [ubuntu-focal](https://github.com/mathworks/build-glibc-bz-19329-patch/releases/tag/ubuntu-focal) (version 9.9) and ran the following command:

```
$ sudo dpkg -i libc6_2.31-0ubuntu9.9.ubuntu.focal.custom_amd64.deb 

[sudo] password for ubuntu: 
(Reading database ... 456624 files and directories currently installed.)
Preparing to unpack libc6_2.31-0ubuntu9.9.ubuntu.focal.custom_amd64.deb ...
De-configuring libc6:i386 (2.31-0ubuntu9.7) ...
Unpacking libc6:amd64 (2.31-0ubuntu9.9.ubuntu.focal.custom) over (2.31-0ubuntu9.7) ...
dpkg: error processing package libc6:amd64 (--install):
 package libc6:amd64 2.31-0ubuntu9.9.ubuntu.focal.custom cannot be configured because libc6:i386 is at a different version (2.31-0ubuntu9.7)
dpkg: error processing package libc6:i386 (--install):
 package libc6:i386 2.31-0ubuntu9.7 cannot be configured because libc6:amd64 is at a different version (2.31-0ubuntu9.9.ubuntu.focal.custom)
Errors were encountered while processing:
 libc6:amd64
 libc6:i386
```

After running the previous command and getting those errors, `$ dpkg-query --show libc6:amd64` gives the following:
```
libc6:amd64	2.31-0ubuntu9.9.ubuntu.focal.custom
```
**SIMULINK is running smoothly**, but I can't install some packages through `apt`. For example, I get errors like this:

```
...
The following packages have unmet dependencies:
 libc6 : Breaks: libc6:i386 (!= 2.31-0ubuntu9.9.ubuntu.focal.custom) but 2.31-0ubuntu9.7 is to be installed
 libc6:i386 : Breaks: libc6 (!= 2.31-0ubuntu9.7) but 2.31-0ubuntu9.9.ubuntu.focal.custom is to be installed
 libc6-dbg : Depends: libc6 (= 2.31-0ubuntu9.7) but 2.31-0ubuntu9.9.ubuntu.focal.custom is to be installed
 libc6-dev : Depends: libc6 (= 2.31-0ubuntu9.7) but 2.31-0ubuntu9.9.ubuntu.focal.custom is to be installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
```

Is there any way I could fix this?

**Comment by josmartin** [comment]
@S4JJ4D What you will probably need to do is ALSO install the pre-built versions of `libc6:i386`, `libc6-dbg`, etc. Once you install one of these (i.e. the original `libc6`) all the others will need to match. One possible way to do this is to download the `tar.gz` of all packages and install all of them in one go with `dpkg`

To be completely honest I don't know what will happen against ubuntu9.7 (rather than ubuntu9.9) - this is the focal-security "pocket" rather than the focal-update "pocket" and doesn't acquire other possible updates from ubuntu. It looks like the version strings match but I can't tell you what patches are applied in "security" as opposed to "update" and there may be drift between these.

**Comment by S4JJ4D** [comment]
Thanks for your reply.

I installed all pre-built packages and then ran `apt --fix-broken install` in the hope that it does some clean-up. I also did some other tinkering (I should have been more careful) after I got errors when I was trying to install some other new tools using `apt`. It seemed to have resolved the issue but I could no longer boot into ubuntu after I restarted the system.

Anyway, I erased my ubuntu partition and installed ubuntu 22.04.
